src/Server/MySQLHandler.cpp

```diff
@@ -371,6 +384,81 @@ void MySQLHandler::comQuery(ReadBuffer & payload)
     }
 }
 
+void MySQLHandler::comStmtPrepare(DB::ReadBuffer & payload)
+{
+    String statement;
+    readStringUntilEOF(statement, payload);
+    auto [is_success, statement_id] = emplacePreparedStatement(std::move(statement));
+    if (is_success)
+        packet_endpoint->sendPacket(PreparedStatementResponseOK(statement_id, 0, 0, 0), true);
+    else
+        packet_endpoint->sendPacket(ERRPacket(), true);
+}
+
+void MySQLHandler::comStmtExecute(ReadBuffer & payload)
+{
+    uint32_t statement_id;
+    payload.readStrict(reinterpret_cast<char *>(&statement_id), 4);
+
+    if (!prepared_statements.contains(statement_id))
```

Comment:
`prepared_statements` must safely only be used under the mutex. My other comment suggests TSA. It would make the compiler scream and cry about this.